### PR TITLE
FIX introduce predict_score instead of transform for learners

### PR DESCRIPTION
**Pull request by moreymat**
Learners provide scores for edges through a new method, `predict_score`, rather than the previously used `transform` which has a different semantics in sklearn's API.

Sklearn classifiers provide a few methods, among which we routinely use:
* `predict`,
* `predict_proba`,
* `decision_function`
Sklearn classifiers also provide a `transform` method (from a mixin) which reduces the input to its most important features, cf. the doc for [LogisticRegression](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#sklearn.linear_model.LogisticRegression.transform) .
The `transform` methods in `attelo.learning` effectively provide scores, like the `predict_proba` and `decision_function` methode they call under the hood.

### CODE REVIEW

**Comment by kowey** [comment]
Looks good to me.  I'd be curious to see the scikit design doc you showed me.

Also do we care that attach/label have different result types? Attach is just a 1-D array (attach scores for all edges).  Labels is scores for all edges times all labels.

I'm sorry I changed the naming from scores to weights (in DataPack.Graph). I was trying to avoid overloading with 'evaluation' (as in "F score").  We need more words.

**Comment by moreymat** [comment]
It seems fine to return either 1-D or 2-D arrays where it is sensible, at least it is done [quite often in sklearn](http://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html#sklearn.linear_model.LogisticRegression.decision_function) .

I am not really concerned about the naming change from scores to weights, it seems quite harmless (at least for the moment).

**Comment by moreymat** [comment]
Here is the reference of the paper you mention:
http://arxiv.org/abs/1309.0238
